### PR TITLE
Update Documentation for Internal Metric Naming Conventions

### LINKED ISSUE #4933: Clarify Collector internal metric names by version
**What needs to be changed?** With release v0.106.1, the names of Collector internal metrics are handled differently, depending on whether they are generated from Collector components (prefixed with `otelcol_`) or instrumentation libraries (not prefixed with `otelcol_`). 

The [list of metrics](https://opentelemetry.io/docs/collector/internal-telemetry/#lists-of-internal-metrics) is correct as-is, but we should add a note that clarifies that versions of the Collector prior to v0.106.1 append the `otelcol_` prefix to all internal metrics, both instrumentation library-generated as well as component-generated.

**What is the name + path of the page that needs changed?** https://opentelemetry.io/docs/collector/internal-telemetry/#lists-of-internal-metrics

**Additional context**: See [comment](https://github.com/open-telemetry/opentelemetry.io/pull/4867#discussion_r1700626960).

### PR DESCRIPTION
**Pull request by Tharanishwaran**
This PR updates the documentation to clarify the internal metric naming conventions for the OpenTelemetry Collector.

Changes:
              Added a note to explain:
              1.Starting from v0.106.1: Metrics from Collector components are prefixed with otelcol_, while metrics from 
                  instrumentation  libraries are not prefixed.
              
              2.Versions before v0.106.1: All internal metrics, regardless of their source, had the otelcol_ prefix.

This update ensures that users understand the changes in metric naming conventions across different versions of the Collector.

Related Issue:
Fixes #4933

### CODE REVIEW

File: content/en/docs/collector/internal-telemetry.md
Comment (tiffany76) [inline content/en/docs/collector/internal-telemetry.md:56]:
```suggestion
{{% alert title="Note" color="info" %}}
As of Collector v0.106.1, internal metric names are handled differently based on their source:
- Metrics generated from Collector components are prefixed with `otelcol_`.
- Metrics generated from instrumentation libraries do not use the `otelcol_` prefix.

For Collector versions prior to v0.106.1, all internal metrics, regardless of their origin, are prefixed with `otelcol_`. This includes metrics from both Collector components and instrumentation libraries.
{{% /alert %}}
```

File: content/en/docs/collector/internal-telemetry.md
Comment (tiffany76) [inline content/en/docs/collector/internal-telemetry.md:66]: We'll need these back ticks to close the yaml code block. Could you add them again?

**Review by tiffany76 — CHANGES_REQUESTED** [review]
Great job! I added short code formatting for the call-out to match others on the page.

I also think this note would be better placed further down where the list of metrics occurs. Right about [line 189](https://github.com/open-telemetry/opentelemetry.io/blob/473fe06d36d72eaa9e4196ff9bcf5b0499d05ddb/content/en/docs/collector/internal-telemetry.md?plain=1#L189) should do it.

**Comment by Tharanishwaran** [comment]
Hi @tiffany76,
I have realized that some unintended commits were included in this pull request. I apologize for the oversight and any inconvenience this may have caused. I will address this issue promptly by removing the extraneous commits and ensuring that only the intended changes remain.

**Review by tiffany76 — APPROVED** [review]

**Comment by tiffany76** [comment]
LGTM, @Tharanishwaran! Thanks for your contribution.

@open-telemetry/collector-approvers, PTAL.

File: content/en/docs/collector/internal-telemetry.md
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:197]:
```suggestion
For Collector versions prior to v0.106.1, all internal metrics emitted via the Prometheus exporter, regardless of
```

**Review by codeboten — COMMENTED** [review]
Thanks @Tharanishwaran! Added a note, the prefix was not applied to internal metrics emitted via OTLP

**Comment by Tharanishwaran** [comment]
Thanks @tiffany76 and @codeboten for the reviews and feedback! The checks have passed and the changes have been approved. Could someone please merge the PR when possible

**Comment by Tharanishwaran** [comment]
Many thanks to @tiffany76, @codeboten  for your useful comments and taking the time to review the changes. I am glad that the checks passed and the PR finally got approval. If possible, one of the maintainers kindly goes ahead and merges the PR. Truly I appreciate the time and your support

**Comment by songy23** [comment]
Thank you @Tharanishwaran  for calling it out, we have run into the same issue in 0.106 & 0.107

**Comment by Tharanishwaran** [comment]
Thank you @songy23 for your  comment I’ve implemented the suggested change regarding the otelcol_ prefix

**Review by songy23 — APPROVED** [review]

File: content/en/docs/collector/internal-telemetry.md
Comment (tiffany76) [inline content/en/docs/collector/internal-telemetry.md:197]:
```suggestion
For Collector versions prior to v0.106.1, all internal metrics emitted using the Prometheus exporter, regardless of
```

**Review by tiffany76 — COMMENTED** [review]
One tiny change.

**Pull request merged**
